spmd/tensor/ops/math_ops.py

```diff
@@ -13,16 +19,14 @@ def dist_sum(self: Tensor) -> Tensor:
 
     local_sum = self_local.sum()
 
-    if isinstance(self_placement, Shard) or isinstance(
-        self_placement, _Partial
-    ):
+    if is_shard(self_placement) or is_partial(self_placement):
```

Comment:
Does it make sense to move `is_shard`, `is_partial` to methods of `Placement`?

Reply:
Yeah that's a good idea, I thought `Placement` is the base type, was wondering if there might be class undefined issue, but I guess I can try it.